manifests/init.pp

```diff
@@ -24,16 +29,26 @@
 # Dan Sajner <dsajner@covermymeds.com>
 #
 class redis (
-  $config             = {},
-  $manage_persistence = false,
-  $slaveof            = undef,
-  $version            = 'installed',
-  $packages           = ['redis']
+  $config               = {},
+  $manage_persistence   = false,
+  $slaveof              = undef,
+  $packages             = ['redis'],
+  String $protected     = 'no',
+  String $redis_conf    = '/etc/redis.conf',
+  String $redis_service = 'redis',
+  String $version       = 'installed',
 ) {
 
   # Install the redis package
   ensure_packages($packages, { 'ensure' => $version })
 
+  # We need to see what version is in use to see if protected-mode should be set.
+  if ( $version =~ /^3.2.*/ ) or ( $version == 'installed' ) {
```

Comment:
There's a dedicated function that's recommended for version comparisons:

https://docs.puppet.com/puppet/4.2/function.html#versioncmp